### DISCUSSION TITLE
How to generate a graph of node-to-node reachability for zwavejs2mqtt

### DISCUSSION THREAD

**Discussion by kdknigga** [zwave-js/zwave-js-ui#1903] [Show and tell]
As a zwavejs2mqtt user, I was a little envious of [Hubitat's topology graph](https://community.inovelli.com/t/poor-signal-w-inovelli/9334?u=krisk), so I decided to replicate it for myself.

And I've decided to make the (simple) source code available to anyone else who'd like to try it:

https://github.com/kdknigga/zwavejs_neighbor_graph

Just ask zwavejs2mqtt to dump your node information (zwavejs2mqtt Control Panel -> Actions -> Advanced -> Dump -> Export) to a json file and feed it into the script.  Some experience with python is probably suggested.

**Comment by robertsLando** [comment]
@kdknigga Thanks!

**Comment by LordMike** [comment]
For those interested, the Zensys Z-Wave PC Controller 5 also has a topology map. I still don't know what to use it for .. but hey :)

![image](https://user-images.githubusercontent.com/1027111/147959673-61bc729c-3ca6-44a3-ba9d-c16209f9ee58.png)
